api/v2/api_test.go

```diff
@@ -1832,6 +1833,51 @@ func Test_API_Initialize_Main_Network(t *testing.T) {
 	api.Close()
 }
 
+func Test_API_Initialize_ListenAndServe(t *testing.T) {
+	// load configuration
+	cfg, err := config.LoadConfig("../../testenv/config.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// setup fake mock clients
+	fakeLens := &mocks.FakeIndexerAPIClient{}
+	fakeOrch := &mocks.FakeServiceClient{}
+	fakeSigner := &mocks.FakeSignerClient{}
```

Comment:
this should be in each test case, as declaring outside the scope of a test case means test cases can interfere with each other and thus are no longer isolated

Reply:
e.g. https://github.com/RTradeLtd/Lens/blob/master/lens_test.go#L71

Reply:
Each test case in this function reuses the same mocks (see line 1861), which we shouldn’t be doing, since it potentially pollutes cases, especially when you test for specific behaviours

Reply:
https://github.com/RTradeLtd/Temporal/pull/270/files/236fae1d3ea06ec0ea7965bf5f649c964132372a..f057a05789484ad6f870f163f9576622c029dd66#diff-6e33e0730d3cb4347a70c7d38ce08ef3R1861

Reply:
I see what you're saying, misread the comment.